### PR TITLE
storage,ingest: enable ingestion optimization using flushableIngest

### PR DESCRIPTION
**Pull request by aadityasondhi**
In https://github.com/cockroachdb/pebble/pull/3398, we introduced an optimization in pebble where we could use flushableIngests with excises for ingesting SSTs with RangeDels and RangeKeyDels.

This patch turns this optimization on using
`sstContainsExciseTombstone`.

Informs https://github.com/cockroachdb/pebble/issues/3335.

Release note: None

### CODE REVIEW

**Comment by aadityasondhi** [comment]
First commit is a pebble bump which will be merged separately. It contained changes that are required for this.

File: pkg/storage/engine.go [hunk 1]
```diff
@@ -1033,11 +1033,22 @@ type Engine interface {
...
-	// IngestAndExciseFiles is a variant of IngestLocalFilesWithStats
-	// that excises an ExciseSpan, and ingests either local or shared sstables or
-	// both.
+	// IngestAndExciseFiles is a variant of IngestLocalFilesWithStats that excises
+	// an ExciseSpan, and ingests either local or shared sstables or both. It also
+	// takes the flag sstsContainExciseTombstone to signal that the exciseSpan
+	// contains RANGEDELs and RANGEKEYDELs.
+	//
+	// NB: It is the callers responsibility to ensure if
+	// sstsContainExciseTombstone is set to true, the snapshot must contain a
```
Comment (itsbilal) [inline pkg/storage/engine.go:1042]: terminology creep; in the storage package, `snapshot` has a different meaning. Say "ingestion sstables".

**Review by itsbilal — APPROVED** [review]
LGTM save for one minor comment

File: pkg/storage/engine.go (hunk 1 above)
Comment (aadityasondhi) [reply pkg/storage/engine.go]: nice catch, fixed! we do overload the term snapshot (pun intended) 😄

**Review by aadityasondhi — COMMENTED** [review]
bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121138#-:-Nu-_Haw3ybvKxbwYWay:b-ydi9ay)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @itsbilal)

**Pull request merged**